### PR TITLE
Add exercice 03

### PR DESCRIPTION
**Pull request by tychota**

### CODE REVIEW

File: content/tutorial-react-native/rn-03.md [hunk 1]
```diff
@@ -0,0 +1,343 @@
+# Tutorial 03 - Advanced Queries
+
+This is the third exercise in the **React Native Track** of this Apollo Client Tutorial! 
+If you prefer React or Angular 2 over React, head over to the respective tutorial track.
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `If you prefer React or Angular 2 over React` => `If you prefer React or Angular 2 over React Native`

File: content/tutorial-react-native/rn-03.md [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+## Goal
+
+In this exercise we will have a look at advanced query features with the **goal** of showing a list of pokemon that our 
+trainer owns. Upon clicking on a pokemon in this list, we will see a detailed view of that pokemon and can navigate back
+to the complete pokedex.
+
+## Introduction
+
+Move to the third exercise of the Pokedex React App and install the dependencies from your console
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `Pokedex React App` => `Pokedex React Native App`

File: content/tutorial-react-native/rn-03.md [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+```
+
+Note that we have to denote the variable type as well, `String!` signifying a required String in this case. Of course, 
+now we also have to supply a value for that variable when we use it to wrap the `Pokedex` component:
+
+```js
+export const TitleWithData = graphql(TrainerQuery, {
+options: {
+    variables: {
+      name: "Tycho"
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `name: "Tycho"` => `name: "__NAME__"`

File: content/tutorial-react-native/rn-03.md [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    }
+  }
+})(Title)
+```
+
+### Nested Queries
+
+Now that we saw query variables in action we can focus on displaying the pokemons of a given trainer. We will use the a 
+React Native ListView as it is the most pragmatic way to display a list of element in a movile app, while being
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `to display a list of element in a movile app` => `to display a list of elements in a mobile app`

File: content/tutorial-react-native/rn-03.md [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+      <View>
+        {this.props.data.Trainer.ownedPokemons.map((pokemon) =>
+          <PokemonListItem key={pokemon.id} pokemon={pokemon} />
+        )}
+      </View>
+    )
+  }
+```
+
+we are going to use react native ListView that create separator between the row, and only render the visible components,
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: work on language

File: content/tutorial-react-native/rn-03.md [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+recycling the old unvisible components to prevent excessive element creation. In other words, a fluid (#60 FPS) scroll view.
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: work on language

File: content/tutorial-react-native/rn-03.md [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+  _renderSeparator = (sectionID, rowID) => (
+    <View
+      key={`${sectionID}-${rowID}`}
+      style={{ height: 1, backgroundColor: '#CCCCCC'}}
+    />
+  );
+}
+```
+
+You can then modify `Title` to accept `this.props;data.Trainer` as props `trainer` and use of
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `this.props;data.Trainer` => `this.props.data.Trainer`

File: content/tutorial-react-native/rn-03.md [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+
+You can then modify `Title` to accept `this.props;data.Trainer` as props `trainer` and use of 
+`this.props.trainer.ownedPokemons.length` that displays the correct amount of pokemons.
+
+To confirm your environment is all correctly setup, start the app now from XDE.
+
+
+### Dynamic Query Variables
+
+If everything working, we can now continue to implement a detailed view of a single pokemon when we click on a
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `If everything working` => `If everything is working`

File: content/tutorial-react-native/rn-03.md [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+To confirm your environment is all correctly setup, start the app now from XDE.
+
+
+### Dynamic Query Variables
+
+If everything working, we can now continue to implement a detailed view of a single pokemon when we click on a 
+`PokemonListItem` component. 
+
+Have a look at the `PokemonDetailScreen` directory that we prepared for you. 
+The component in `screen/PokemonDetailScreen/components/PokemonDetail.js` render the screen based on the props pokemon
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `render` => `renders`

File: content/tutorial-react-native/rn-03.md [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+
+
+### Dynamic Query Variables
+
+If everything working, we can now continue to implement a detailed view of a single pokemon when we click on a 
+`PokemonListItem` component. 
+
+Have a look at the `PokemonDetailScreen` directory that we prepared for you. 
+The component in `screen/PokemonDetailScreen/components/PokemonDetail.js` render the screen based on the props pokemon 
+that the page pass.
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: that the page passes?

what is the page? maybe clarify

File: content/tutorial-react-native/rn-03.md [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+
+We could have use the existing data from the list to rendre the whole screen, without an extra request but we will still
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `We could have use the existing data from the list to rendre the whole screen, without an extra request but we will still`
=> `We could have used the existing data from the list to render the whole screen without an extra request, but we will still`

File: content/tutorial-react-native/rn-03.md [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+do the extra request to show yout how to customize the request based on the route.
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `yout` => `yout`

File: content/tutorial-react-native/rn-03.md [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+
+We already created the navigation in `navigation/` that assigns the `PokemonDetailScreen` to the path `pokemonDetail` 
+so we can. Exponent navigation allows us to pass extra parameters to the route like this :
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `so we can.` ??

File: content/tutorial-react-native/rn-03.md [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+```js
+_goToDetail = (pokemon) => {
+  this.props.navigation
+    .getNavigator('root')
+    .push(Router.getRoute('pokemonDetail', { pokemon }));
+}
+```
+
+where `getNavigator` allows to select the root navigator (and not the tab navigator) and push some routes on it, which makes
+the second order pages'tab bar disappear, and `push` allow to pass an extra object of parameters.
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `pages'tab bar`??

Comment (marktani) [reply content/tutorial-react-native/rn-03.md]: `and`push`allow to pass` => `and`push`allows to pass`

File: content/tutorial-react-native/rn-03.md [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+
+Then we can get the paramaters like this :
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `this :` => `this:`

File: content/tutorial-react-native/rn-03.md [hunk 16]
```diff
@@ -0,0 +1,343 @@
...
+      id
+      url
+      name
+    }
+  }
+`
+```
+
+As you can see, the query requires a query variable `id` of type `ID` that we have to supply using the query option 
+`variables` as before. However, other than with the trainer name variable, we cannot just use a cannot value as the
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: `we cannot just use a cannot value` => `we cannot just use a value`

File: content/tutorial-react-native/rn-03.md [hunk 17]
```diff
@@ -0,0 +1,343 @@
...
+    return {
+      variables: {
+        id: ownProps.pokemonId
+      }
+    }
+  }
+})(PokemonDetail);
+```
+
+Note that by default Apollo try to fill unfilled `variable` with `ownProps.variable`, so this was valid too:
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md]: Note that by default Apollo tries to fill any unfilled variable called `variable` with `ownProps.variable`, so this is valid too

File: content/tutorial-react-native/rn-03.md [hunk 18]
```diff
@@ -0,0 +1,343 @@
...
+Click the back button on the nav bar or the native back button on Android to go back to your pokedex.
+
+## Recap
+
+Nice, our pokedex starts to get some shape! We added both an overview of all our pokemons as well as a detailed view of 
+a single pokemon. Let's go through what we saw in this exercise again:
+
+* **Constants as query variables** helped us getting started with advanced query features
+* Using GraphQL, we can easily create **nested queries**
+* Combining **router parameters** and **dynamic query variables**, we were able to supply different data to the same
```
Comment (marktani) [inline content/tutorial-react-native/rn-03.md:351]: is this still relevant? maybe customize this line to the exponent navigation stuff

File: content/tutorial-react-native/rn-03.md (hunk 14 above)
Comment (tychota) [reply content/tutorial-react-native/rn-03.md]: The idea I want to express is "native app usually have a tab bar only in the first layer. As soon as you click one item in the page, the detail page have no tab bar." But indeed that has no sense.

**Review by marktani — APPROVED** [review]
LGTM

**Pull request merged**